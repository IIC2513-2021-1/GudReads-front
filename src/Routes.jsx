import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from './views/Home';
import AuthorList from './views/AuthorList';
import AuthorDetail from './views/AuthorDetail';
import NotFound from './views/NotFound';
import Login from './views/Login';
import AuthContextProvider from './contexts/AuthContext';

export default function Routes() {
  return (
    <Router>
      <AuthContextProvider>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/login" component={Login} />
          <Route exact path="/authors" component={AuthorList} />
          <Route exact path="/authors/:id" component={AuthorDetail} />
          <Route component={NotFound} />
        </Switch>
      </AuthContextProvider>
    </Router>
  );
}
