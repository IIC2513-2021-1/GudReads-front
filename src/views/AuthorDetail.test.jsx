import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Routes from '../Routes';

const { act } = renderer;

function TestRouter({ path }) {
  return (
    <MemoryRouter initialEntries={[path]}>
      <Routes />
    </MemoryRouter>
  );
}

const authorResponse = {
  data: {
    type: 'authors',
    id: '1',
    attributes: {
      firstName: 'Joanne',
      lastName: 'Rowling',
      birthDate: '1965-07-31',
      imageUrl: 'https://res.cloudinary.com/dee8jw8ue/image/upload/v1623992961/descarga_puc6lu.jpg',
    },
  },
};

describe('AuthorDetail', () => {
  describe('when user is not logged in', () => {
    beforeEach(() => {
      fetch.resetMocks();
    });

    it('renders the author data along with their books', async () => {
      let testRenderer;

      fetch.mockResponse(JSON.stringify(authorResponse));
      // After first render and useEffect execution
      // Which in turn calls setLoading(true)
      // Therefore it triggers a second render
      act(() => {
        testRenderer = renderer.create(<TestRouter path="/authors/1" />);
      });

      // fetch is async so we need to wait for it to finish
      // No need to write a body for the function inside act
      await act(async () => {});

      const tree = testRenderer.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
