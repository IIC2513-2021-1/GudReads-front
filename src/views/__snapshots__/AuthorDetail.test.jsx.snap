// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorDetail when user is not logged in renders the author data along with their books 1`] = `
<div>
  <a
    href="/authors"
    onClick={[Function]}
  >
    Authors
  </a>
  <div>
    <h2>
      Joanne Rowling
    </h2>
    <p>
      The author was born in 1965-07-31
    </p>
    <img
      alt="Joanne Rowling"
      src="https://res.cloudinary.com/dee8jw8ue/image/upload/v1623992961/descarga_puc6lu.jpg"
    />
    <div
      className="books"
    >
      <div
        className="title"
      >
        <h3>
          Books
        </h3>
      </div>
      <ul
        className="book-list"
      >
        <li
          className="single-book"
        >
          <h3>
            Libro 1
          </h3>
          <img
            alt="book"
            src="https://d1csarkz8obe9u.cloudfront.net/posterpreviews/action-thriller-book-cover-design-template-3675ae3e3ac7ee095fc793ab61b812cc_screen.jpg?ts=1588152105"
          />
        </li>
        <li
          className="single-book"
        >
          <h3>
            Libro 2
          </h3>
          <img
            alt="book"
            src="https://d1csarkz8obe9u.cloudfront.net/posterpreviews/classic-novel-kindle-book-cover-design-template-59f47ff5c79cbd1d5fa0c14e585217eb_screen.jpg?ts=1561444051"
          />
        </li>
        <li
          className="single-book"
        >
          <h3>
            Libro 3
          </h3>
          <img
            alt="book"
            src="https://d1csarkz8obe9u.cloudfront.net/posterpreviews/contemporary-fiction-night-time-book-cover-design-template-1be47835c3058eb42211574e0c4ed8bf_screen.jpg?ts=1594616847"
          />
        </li>
        <li
          className="single-book"
        >
          <h3>
            Libro 4
          </h3>
          <img
            alt="book"
            src="https://d1csarkz8obe9u.cloudfront.net/posterpreviews/thriller-fantasy-book-cover-castle-mountain-design-template-890a4e62a2009241b05e70a59186240d_screen.jpg?ts=1599683486"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
